lstchain/scripts/onsite/onsite_create_calibration_file.py

```diff
@@ -98,198 +98,206 @@ def main():
+
+    # search the summary file info
+    run_summary_path = f"{base_dir}/monitoring/RunSummary/RunSummary_{date}.ecsv"
+    if not os.path.exists(run_summary_path):
+        print(f"Night summary file {run_summary_path} does not exist")
+        sys.exit(3)
+
+    print(f"\n--> Use run summary {run_summary_path}")
+    # pedestal base dir
+    ped_dir = f"{base_dir}/monitoring/PixelCalibration/drs4_baseline/"
+
+    # search the pedestal file of the same date
+    if ped_run is None:
+        # else search the pedestal file of the same date
+        file_list = sorted(Path(f"{ped_dir}/{date}/{prod_id}/").rglob('drs4_pedestal*.0000.fits'))
         if len(file_list) == 0:
-            raise IOError(f"Run {run} not found\n")
+            print(f"No pedestal file found for date {date}")
+            sys.exit(4)
+        if len(file_list) > 1:
+            print(f"Too many pedestal files found for date {date}: {file_list}, choose one run")
+            sys.exit(5)
         else:
-            input_file = file_list[0]
-        print(f"\n--> Input file: {input_file}")
-
-        # find date
-        input_dir, name = os.path.split(os.path.abspath(input_file))
-        path, date = input_dir.rsplit('/', 1)
-
-        # verify output dir
-        output_dir = f"{base_dir}/monitoring/PixelCalibration/calibration/{date}/{prod_id}"
-        if not os.path.exists(output_dir):
-            print(f"--> Create directory {output_dir}")
-            os.makedirs(output_dir, exist_ok=True)
-
-        # make log dir
-        log_dir = f"{output_dir}/log"
-        if not os.path.exists(log_dir):
-            print(f"--> Create directory {log_dir}")
-            os.makedirs(log_dir, exist_ok=True)
-
-        # search the summary file info
-        run_summary_path = f"{base_dir}/monitoring/RunSummary/RunSummary_{date}.ecsv"
-        if not os.path.exists(run_summary_path):
-            raise IOError(f"Night summary file {run_summary_path} does not exist\n")
-
-        print(f"\n--> Use run summary {run_summary_path}")
-        # pedestal base dir
-        ped_dir = f"{base_dir}/monitoring/PixelCalibration/drs4_baseline/"
-
-        # search the pedestal file of the same date
-        if ped_run is None:
-            # else search the pedestal file of the same date
-            file_list = sorted(Path(f"{ped_dir}/{date}/{prod_id}/").rglob('drs4_pedestal*.0000.fits'))
-            if len(file_list) == 0:
-                raise IOError(f"No pedestal file found for date {date}\n")
-            if len(file_list) > 1:
-                raise IOError(f"Too many pedestal files found for date {date}: {file_list}, choose one run\n")
-            else:
-                pedestal_file = file_list[0]
+            pedestal_file = file_list[0]
 
-        # else, if given, search a specific pedestal run
+    # else, if given, search a specific pedestal run
+    else:
+        file_list = sorted(Path(f"{ped_dir}").rglob(f'*/{prod_id}/drs4_pedestal.Run{ped_run}.0000.fits'))
```

Comment:
I think that in the first `Path` definition, the fstring can be avoided:
`file_list = sorted(Path(ped_dir).rglob(f'*/{prod_id}/drs4_pedestal.Run{ped_run}.0000.fits'))`